### PR TITLE
Sourcery Starbot ⭐ refactored manncodes/xeno

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>master</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/xeno master
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: xeno/activations.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def forward(self, input):
         return np.maximum(0.0, input)
 
     def derivative(self, input=None):
-        last_forward = input if input else self.last_forward
+        last_forward = input or self.last_forward
```
Comment (SourceryAI) [inline xeno/activations.py:62]: Function `ReLU.derivative` refactored with the following changes:
- Simplify if expression by using or ([`or-if-exp-identity`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#or-if-exp-identity))

File: xeno/activations.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def forward(self, input):
         return input
 
     def derivative(self, input=None):
-        last_forward = input if input else self.last_forward
+        last_forward = input or self.last_forward
```
Comment (SourceryAI) [inline xeno/activations.py:79]: Function `Linear.derivative` refactored with the following changes:
- Simplify if expression by using or ([`or-if-exp-identity`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#or-if-exp-identity))

File: xeno/activations.py [hunk 3]
```diff
@@ -93,11 +93,10 @@ def forward(self, input):
         self.last_forward = input
         x = input - np.max(input, axis=1, keepdims=True)
         exp_x = np.exp(x)
-        s = exp_x / np.sum(exp_x, axis=1, keepdims=True)
-        return s
+        return exp_x / np.sum(exp_x, axis=1, keepdims=True)
```
Comment (SourceryAI) [inline xeno/activations.py:96]: Function `Softmax.forward` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: xeno/activations.py [hunk 4]
```diff
@@ -93,11 +93,10 @@ def forward(self, input):
...
 
     def derivative(self, input=None):
-        last_forward = input if input else self.last_forward
+        last_forward = input or self.last_forward
```
Comment (SourceryAI) [inline xeno/activations.py:99]: Function `Softmax.derivative` refactored with the following changes:
- Simplify if expression by using or ([`or-if-exp-identity`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#or-if-exp-identity))

File: xeno/initializations.py [hunk 5]
```diff
@@ -95,7 +95,7 @@ def decompose_size(size):
         fan_in = size[0]
         fan_out = size[1]
 
-    elif len(size) == 4 or len(size) == 5:
+    elif len(size) in [4, 5]:
```
Comment (SourceryAI) [inline xeno/initializations.py:98]: Function `decompose_size` refactored with the following changes:
- Replace multiple comparisons of same variable with `in` operator ([`merge-comparisons`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#merge-comparisons))

File: xeno/model.py [hunk 6]
```diff
@@ -64,10 +64,7 @@ def fit(self, X, Y, max_iter=100, batch_size=64, shuffle=True,
         else:
             valid_X, valid_Y = None, None
 
-        iter_idx = 0
-        while iter_idx < max_iter:
-            iter_idx += 1
-
+        for iter_idx in range(1, max_iter + 1):
```
Comment (SourceryAI) [inline xeno/model.py:67]: Function `Model.fit` refactored with the following changes:
- Replace while with for ([`while-to-for`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#while-to-for))

File: xeno/model.py [hunk 7]
```diff
@@ -142,8 +139,7 @@ def predict(self, X):
         x_next = X
         for layer in self.layers[:]:
             x_next = layer.forward(x_next)
-        y_pred = x_next
-        return y_pred
+        return x_next
```
Comment (SourceryAI) [inline xeno/model.py:142]: Function `Model.predict` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: xeno/layers/core.py [hunk 8]
```diff
@@ -86,8 +86,7 @@ def forward(self, input, *args, **kwargs):
         
         self.last_input = input
         linear_out = np.dot(input, self.W) + self.b
-        act_out = self.act_layer.forward(linear_out)
-        return act_out
+        return self.act_layer.forward(linear_out)
```
Comment (SourceryAI) [inline xeno/layers/core.py:89]: Function `Dense.forward` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

File: xeno/layers/core.py [hunk 9]
```diff
@@ -128,11 +127,10 @@ def connect_to(self, prev_layer):
...
         
         if 0. < self.p < 1.:
-            if train:
-                self.last_mask = get_rng().binomial(1, 1 - self.p, input.shape) / (1 - self.p)
-                return input * self.last_mask
-            else:
+            if not train:
                 return input * (1 - self.p)
+            self.last_mask = get_rng().binomial(1, 1 - self.p, input.shape) / (1 - self.p)
+            return input * self.last_mask
```
Comment (SourceryAI) [inline xeno/layers/core.py:133]: Function `Dropout.forward` refactored with the following changes:
- Swap if/else branches ([`swap-if-else-branches`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#swap-if-else-branches))
- Remove unnecessary else after guard condition ([`remove-unnecessary-else`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#remove-unnecessary-else))

File: xeno/layers/normalization.py [hunk 10]
```diff
@@ -92,10 +92,7 @@ def backward(self, pre_grad, *args, **kwargs):
         dmu = -1 * np.sum(dxmu1 + dxmu2, axis=0)
         dx2 = 1. / N * np.ones((N, D)) * dmu
 
-        # step0 done!
-        dx = dx1 + dx2
-
-        return dx
+        return dx1 + dx2
```
Comment (SourceryAI) [inline xeno/layers/normalization.py:95]: Function `BatchNormal.backward` refactored with the following changes:
- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://github.com/sourcery-ai/sourcery/wiki/Current-Refactorings#inline-immediately-returned-variable))

This removes the following comments ( [why?](https://github.com/sourcery-ai/sourcery/wiki/FAQ#comment-removal) ):
```
# step0 done!
```
